src/main/java/com/camunda/consulting/simulator/property/CachedPropertyUtil.java

```diff
@@ -0,0 +1,50 @@
+package com.camunda.consulting.simulator.property;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import org.camunda.bpm.engine.delegate.DelegateExecution;
+import org.camunda.bpm.engine.impl.el.Expression;
+import org.camunda.bpm.model.bpmn.instance.BaseElement;
+import org.camunda.bpm.model.xml.instance.ModelElementInstance;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.camunda.consulting.simulator.SimulatorPlugin;
+
+public class CachedPropertyUtil {
```

Comment:
I would put all that in the ModelPropertyUtil. Because: does it ever make sense to not cache a property? So I would also not use the word "cached" in class or method names.

Reply:
It is ok to not do it in this PR but create a chore ticket for that.

Reply:
Will be handled in #3